hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -555,6 +558,26 @@ private void bindAWSClient(URI name, boolean dtEnabled) throws IOException {
         .createS3Client(getUri(), bucket, credentials, uaSuffix);
   }
 
+  /**
+   * Implementation of all operations used by delegation tokens.
+   */
+  private class DelegationOperationsImpl implements DelegationOperations {
```

Comment:
With every commit, I try to factor out a little bit from S3AFs and DDB. I'd like to know if there's a possibility for this to get factored out instead of adding this to the S3AFs class.

Reply:
This is the interface of all operations which the delegation token implementation needs to make of the FS. It's gone in precisely because I've just locked off all access to "the FS". Yes, we do you now have a set of inner classes to the file system, with OperationCallbacksImpl the big one. That is along with WriteOperationHelper, which was only pulled out of S3A FS itself in HADOOP-13786. But what we are moving away from is saying "here is the FS, call any method you want". As we re-fact that the code further, the implementation classes are things we should be able to move further down into my envisaged S3AStore class. (more specifically, into S3AStoreImpl). For now though: inner classes of the file system itself.